src/components/NavBar/SearchBarDropdown.tsx

```diff
@@ -130,24 +134,68 @@ interface SearchBarDropdownProps {
   isLoading: boolean
 }
 
-export const SearchBarDropdown = ({
-  toggleOpen,
-  tokens,
-  collections,
+const useSearchTrendingTokens = () => {
+  const { pathname } = useLocation()
+  const isTokenPage = pathname.includes('/tokens')
+  const { data: trendingTokenData } = useTrendingTokens(useWeb3React().chainId)
+  const trendingTokensLength = isTokenPage ? 3 : 2
+  return {
+    data: trendingTokenData,
+    results: useMemo(
+      () => trendingTokenData?.slice(0, trendingTokensLength) ?? [...Array<SearchToken>(trendingTokensLength)],
+      [trendingTokenData, trendingTokensLength]
+    ),
+  }
+}
+
+export const SearchBarDropdown = (props: SearchBarDropdownProps) => {
+  const { isLoading } = props
+  const { chainId } = useWeb3React()
+  const showBNBComingSoonBadge = chainId === SupportedChainId.BNB && !isLoading
+
+  return (
+    <Column overflow="hidden" className={clsx(styles.searchBarDropdownNft, styles.searchBarScrollable)}>
+      <Box opacity={isLoading ? '0.3' : '1'} transition="125">
+        <SuspenseFallbackToPreviousContents>
+          <Suspend when={isLoading}>
+            <SearchBarDropdownContents {...props} />
+          </Suspend>
+        </SuspenseFallbackToPreviousContents>
+        {showBNBComingSoonBadge && (
+          <BNBComingSoonBadge>
+            <BNBLogo src={BnbLogoURI} />
+            <ThemedText.BodySmall color="textSecondary" fontSize="14px" fontWeight="400" lineHeight="20px">
+              <Trans>Coming soon: search and explore tokens on BNB Chain</Trans>
+            </ThemedText.BodySmall>
+          </BNBComingSoonBadge>
+        )}
+      </Box>
+    </Column>
+  )
+}
+
+const SearchBarDropdownContents = ({
```

Comment:
below here its just the same logic pulled into a component rather than a useEffect, with some minor changes